test/buy-and-make/revenue-split-buy-back.spec.ts

```diff
@@ -0,0 +1,640 @@
+        staking2 = await new MockStakingContract__factory(sa.default.signer).deploy()
+        await staking1.setTotalSupply(simpleToExactAmount(3000000))
+        await staking2.setTotalSupply(simpleToExactAmount(1000000))
+
+        // Deploy mock Nexus
+        nexus = await new MockNexus__factory(sa.default.signer).deploy(
+            sa.governor.address,
+            sa.mockSavingsManager.address,
+            sa.mockInterestValidator.address,
+        )
+        await nexus.setKeeper(sa.keeper.address)
+
+        // Mocked Uniswap V3
+        uniswap = await new MockUniswapV3__factory(sa.default.signer).deploy()
+        // Add rewards to Uniswap
+        await rewardsToken.transfer(uniswap.address, simpleToExactAmount(500000))
+        // Add bAsset to rewards exchange rates
+        await uniswap.setRate(bAsset1.address, rewardsToken.address, simpleToExactAmount(80, 16)) // 0.8 MTA/USD
+        await uniswap.setRate(bAsset2.address, rewardsToken.address, simpleToExactAmount(50, 33)) // 50,000 MTA/BTC
+        // Uniswap paths
+        uniswapMusdBasset1Paths = encodeUniswapPath([bAsset1.address, DEAD_ADDRESS, rewardsToken.address], [3000, 3000])
+        uniswapMbtcBasset2Paths = encodeUniswapPath([bAsset2.address, DEAD_ADDRESS, rewardsToken.address], [3000, 3000])
+
+        // Deploy Emissions Controller
+        const defaultConfig = {
+            A: -166000,
+            B: 180000,
+            C: -180000,
+            D: 166000,
+            EPOCHS: 312,
+        }
+        emissionController = await new EmissionsController__factory(sa.default.signer).deploy(
+            nexus.address,
+            rewardsToken.address,
+            defaultConfig,
+        )
+        await emissionController.initialize(
+            [staking1.address, staking2.address],
+            [10, 10],
+            [true, true],
+            [staking1.address, staking2.address],
+        )
+        await rewardsToken.transfer(emissionController.address, simpleToExactAmount(10000))
+
+        // Deploy and initialize test RevenueSplitBuyBack
+        revenueBuyBack = await new RevenueSplitBuyBack__factory(sa.default.signer).deploy(
+            nexus.address,
+            rewardsToken.address,
+            uniswap.address,
+            emissionController.address,
+        )
+        // reverse the order to make sure dial id != staking contract id for testing purposes
+        await revenueBuyBack.initialize([1, 0], await treasury.getAddress(), _treasuryFee)
+
+        // Add config to buy rewards from mAssets
+        await revenueBuyBack.connect(sa.governor.signer).mapBasset(mUSD.address, bAsset1.address)
+        await revenueBuyBack.connect(sa.governor.signer).mapBasset(mBTC.address, bAsset2.address)
+    }
+
+    before(async () => {
+        const accounts = await ethers.getSigners()
+        mAssetMachine = await new MassetMachine().initAccounts(accounts)
+        sa = mAssetMachine.sa
+
+        await setupRevenueBuyBack(treasuryFee)
```

Comment:
is it necessary to add  await setupRevenueBuyBack(treasuryFee) before all tests? 
I think that it is initialised anyway  within each  internal`describe`